### ISSUE TITLE
EOOO4 error from test script in README.md

### ISSUE THREAD

**Issue by dk-means** [AuthorizeNet/sdk-php#97]
I've just gotten the sdk-php-master package installed on my development server, and tried to run the example script in README.md.  The log file shows the XML sent and the response, but I have no idea what to do next to correct this error.

Any suggestions will be appreciated.

--------------------------------------------------------------------------------------------------------------------------------

 Thu, 10 Dec 2015 00:11:32 +0000 INFO : [_sendRequest] (/home/means/trunk/weddingsite/authorize.net/lib/net/authorize/util/HttpClient.php : 78) -  Url: https://apitest.authorize.net/xml/v1/request.api
 Thu, 10 Dec 2015 00:11:32 +0000 INFO : [_sendRequest] (/home/means/trunk/weddingsite/authorize.net/lib/net/authorize/util/HttpClient.php : 80) - Request to AnetApi:
<?xml version="1.0" encoding="UTF-8"?>
<result>
  <merchant_authentication>
    <name><![CDATA[--obscured--]]></name>
    <transaction_key><![CDATA[--obscured]]></transaction_key>
  </merchant_authentication>
  <transaction_request>
    <transaction_type><![CDATA[authCaptureTransaction]]></transaction_type>
    <amount>151.51</amount>
    <payment>
      <credit_card>
        <card_number><![CDATA[xxxx]]></card_number>
        <expiration_date><![CDATA[2038-12]]></expiration_date>
      </credit_card>
    </payment>
  </transaction_request>
</result>

 Thu, 10 Dec 2015 00:11:32 +0000 INFO : [_sendRequest] (/home/means/trunk/weddingsite/authorize.net/lib/net/authorize/util/HttpClient.php : 92) - Sending 'XML' Request type
 Thu, 10 Dec 2015 00:11:32 +0000 INFO : [_sendRequest] (/home/means/trunk/weddingsite/authorize.net/lib/net/authorize/util/HttpClient.php : 97) - Sending http request via Curl
 Thu, 10 Dec 2015 00:11:33 +0000 INFO : [_sendRequest] (/home/means/trunk/weddingsite/authorize.net/lib/net/authorize/util/HttpClient.php : 99) - Response from AnetApi:  <?xml version="1.0" encoding="utf-8"?><ErrorResponse xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns="AnetApi/xml/v1/schema/AnetApiSchema.xsd"><messages><resultCode>Error</resultCode><message><code>E00004</code><text>The name of the requested API method is invalid.</text></message></messages></ErrorResponse>

**Comment by brianmc** [comment]
Hi @dk-means, can you pull down the sample code repo https://github.com/AuthorizeNet/sample-code-php and try the charge-credit-card.php sample in that?  That will help us isolate the issue a little further.

Thanks,  
Brian

**Comment by dk-means** [comment]
Sorry to report that this makes essentially no difference.  The XML is longer, but the response is the same.  In my initial report, I did not also mention that the script produces an exception after the CURL transaction completes.  I will attach both the log info from the CURL part of the run and the exception report.
[CURL-xml_log.txt](https://github.com/AuthorizeNet/sdk-php/files/58484/CURL-xml_log.txt)
[charge.txt](https://github.com/AuthorizeNet/sdk-php/files/58485/charge.txt)


---------------------------------------------------------------------------------------------------------------------------------

**Comment by brianmc** [comment]
That info really helped, I can see the XML request is indeed invalid as reported in the error.  Have you run "composer update" in the root of the sample-code-php directory?

**Comment by dk-means** [comment]
On 12/10/2015 11:27 AM, Brian McManus wrote:
>
> That info really helped, I can see the XML request is indeed invalid as reported in the error. 
> Have you run "composer update" in the root of the sample-code-php directory?
>
No, I only loaded all this stuff yesterday, and ran composer update then.

**Comment by brianmc** [comment]
Would you mind attaching the output from > php PaymentTransactions/charge-credit-card.php  the files you sent seem to be from a file called charge-card-test.php

**Comment by dk-means** [comment]
On 12/10/2015 11:35 AM, Brian McManus wrote:
>
> Would you mind attaching the output from > php PaymentTransactions/charge-credit-card.php the 
> files you sent seem to be from a file called charge-card-test.php
>
The script charge-card-test.php is a copy of the GitHub script charge-credit-card.php, with my 
credentials substituted for those that came with the original.  I made no other changes to the 
PHP file before running it.

David Means

**Comment by brianmc** [comment]
OK no worries.  I would help us to base line if you could just run it untouched from github.  ie. make sure git status does not show any diffs.  Thanks again,  Brian

**Comment by dk-means** [comment]
Hi Brian.  I finally gave up on the SDK, and began writing PHP to create the XML for my transaction type directly.  I soon discovered that the XML available at developer.authorize.net under "Charge a Credit Card" differed in one essential respect from that I posted earlier (remember that the XML I reported was generated by the SDK, not by me.)  The first tag below <createTransaction Request ...> in the test case on developer.authorize.net is spelled <merchantAuthentication>.  In the XML generated by the SDK, that tag is spelled <merchant_authentication>.  It appears that there is, at minimum, a version skew between the SDK and the endpoint at authorize.net.

I am chucking the entire SDK, both because it is bloated and opaque, and because it is wrong.
